django/university/views.py

```diff
@@ -37,6 +37,12 @@ def course(request, year):
     json_data = list(Course.objects.filter(year=year).values())
     return JsonResponse(json_data, safe=False)
 
+@api_view(['GET'])
+def course_unit_by_id(request, course_unit_id):
+    course_unit = list(CourseUnit.objects.filter(id=course_unit_id).values())
```

Comment:
The list method is necessary? Because we are returning only one item, or it will impact to remove it?

Reply:
You are absolutely right :+1: !

I didn't give it much of a thought while doing it, but now instead of returning a list, since there are only one course unit with that id due to the `UNIQUE` constraint present in our database, now it is calling the `.first()` method on the query set and returning only one. The `first()` method returns `None` if no course unit is found with that id.

I also added a `response/errors.py` folder in order to report a 404 not found error with a custom message for the course unit id not being found (`e.g.  the first() method returns None)

A quick note is that no other endpoint is handling errors by returning the correct HTTP status code. For example, the the `course` endpoint does not return the correct HTTP status code if  the year is not found. This folder is a start of an attempt of introducing that kind of behaviour to our API.

Reply:
Well done! I agree that we can look for that later